src/test/java/org/xstefank/check/RequiredRowsCheckTest.java

```diff
@@ -50,4 +51,19 @@ public void testIfNullParameterThrowsException() {
     public void testIfEmptyPayloadThrowsException() throws IOException {
         requiredRowsCheck.check(TestUtils.EMPTY_PAYLOAD);
     }
+
+    @Test
+    public void testMultipleFailedLines() {
+        RegexDefinition secondRow = new RegexDefinition();
+        String expectedErrorMessage = "Second error message";
+
+        secondRow.setPattern(Pattern.compile("(?!.*)"));
+        secondRow.setMessage(expectedErrorMessage);
+
+        requiredRowsCheck = new RequiredRowsCheck(Arrays.asList(row, secondRow));
+        String errorMessage = requiredRowsCheck.check(TestUtils.TEST_PAYLOAD);
```

Comment:
maybe I would also rename this as something more specific...e.g. actualErrorMessage :)

Reply:
I think that one is clear as it's directly checked on the next lines